core/framework/extractors/docling_extractor_backup_20250913.py

```
@@ -0,0 +1,403 @@
+            For PDF extraction, the exact keys come from the underlying extractor but will include metadata with at least
+            'extractor', 'num_pages' (when available), 'processing_time', and 'pdf_path'.
+        
+        Raises:
+            FileNotFoundError: If the provided path does not exist.
+            RuntimeError: For empty files, invalid PDF header, unreadable files, text read failures, extraction failures,
+                          or when Docling is unavailable and fallback is disabled.
+        """
+        pdf_path = Path(pdf_path)
+        
+        if not pdf_path.exists():
+            raise FileNotFoundError(f"PDF not found: {pdf_path}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)